src/components/register/register.jsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react';
+import { instanceOf } from 'prop-types';
+import { useNavigate } from 'react-router-dom';
+import {
+  FormControl,
+  Image,
+  FormLabel,
+  InputGroup,
+  Input,
+  InputRightElement,
+  Button,
+  Flex,
+  Link,
+} from '@chakra-ui/react';
+import { Cookies, withCookies } from '../../utils/cookie_utils';
+import { registerWithEmailAndPassword, signInWithGoogle } from '../../utils/auth_utils';
+
+const Register = ({ cookies }) => {
+  const [errorMessage, setErrorMessage] = useState();
+  const [firstName, setFirstName] = useState();
+  const [lastName, setLastName] = useState();
+  const [email, setEmail] = useState();
+  const [password, setPassword] = useState();
+  const [checkPassword, setCheckPassword] = useState();
+  const [showPassword, setShowPassword] = useState();
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'setShowPassword' is assigned a value but never used.